File: AdminSistemi.java
import java.util.Date;

public class AdminSistemi {
    private String adminId;
    private String emri;
    private String mbiemri;
    private String email;
    private String fjalëkalimi;
    private String roli = "Admin";
    private int    nivelPrivilegjeve;   // 1=ReadOnly, 2=Standard, 3=Super
    private String gjendjaLlogarise;
    private Date   datëRegjistrimi;

    public AdminSistemi() {}

    public AdminSistemi(String adminId, String emri, String mbiemri,
                        String email, String fjalëkalimi,
                        int nivelPrivilegjeve, String gjendjaLlogarise) {
        this.adminId           = adminId;
        this.emri              = emri;
        this.mbiemri           = mbiemri;
        this.email             = email;
        this.fjalëkalimi       = fjalëkalimi;
        this.nivelPrivilegjeve = nivelPrivilegjeve;
        this.gjendjaLlogarise  = gjendjaLlogarise;
        this.datëRegjistrimi   = new Date();
    }

    public boolean login(String email, String pass) {
        return this.email.equals(email) && this.fjalëkalimi.equals(pass);
    }

    public void bllokIP(String ip)              { System.out.println("IP bllokuar: " + ip); }
    public void riaktivizoLlogari(String userId) { System.out.println("Llogaria riaktivizuar: " + userId); }
    public void konfigurojFirewall()             { System.out.println("Firewall konfiguruar."); }
    public void gjenerojRaport()                 { System.out.println("Raport gjeneruar."); }

    // ── Getters & Setters ──────────────────────────────────────────────────
    public String getAdminId()              { return adminId; }
    public void   setAdminId(String v)       { this.adminId = v; }
    public String getEmri()                 { return emri; }
    public void   setEmri(String v)          { this.emri = v; }
    public String getMbiemri()              { return mbiemri; }
    public void   setMbiemri(String v)       { this.mbiemri = v; }
    public String getEmail()                { return email; }
    public void   setEmail(String v)         { this.email = v; }
    public String getFjalëkalimi()          { return fjalëkalimi; }
    public void   setFjalëkalimi(String v)   { this.fjalëkalimi = v; }
    public String getRoli()                 { return roli; }
    public int    getNivelPrivilegjeve()    { return nivelPrivilegjeve; }
    public void   setNivelPrivilegjeve(int v){ this.nivelPrivilegjeve = v; }
    public String getGjendjaLlogarise()     { return gjendjaLlogarise; }
    public void   setGjendjaLlogarise(String v){ this.gjendjaLlogarise = v; }
    public Date   getDatëRegjistrimi()      { return datëRegjistrimi; }

    public String getNivelLabel() {
        switch(nivelPrivilegjeve) {
            case 1: return "ReadOnly";
            case 2: return "Standard";
            case 3: return "Superadmin";
            default: return "Unknown";
        }
    }

    @Override
    public String toString() {
        return "[" + adminId + "] " + emri + " " + mbiemri + " — Nivel " + nivelPrivilegjeve;
    }
}

File: python/university_security/models/admin_sistemi.py
"""AdminSistemi domain class — mirrors AdminSistemi.java."""

from __future__ import annotations

from dataclasses import dataclass, field
from datetime import datetime

from .actor import Actor


@dataclass
class AdminSistemi(Actor):
    """System administrator account.

    Attributes
    ----------
    admin_id:           Unique identifier (e.g. ``"A001"``).
    emri:               First name.
    mbiemri:            Last name.
    email:              University e-mail address.
    fjalekalimi:        Hashed / plain password (demo only).
    nivel_privilegjeve: Privilege level — ``1`` ReadOnly, ``2`` Standard,
                        ``3`` Superadmin.
    gjendja_llogarise:  Account state — ``"Aktive"``, ``"Locked"``,
                        or ``"UnderReview"``.
    date_regjistrimi:   Registration timestamp (default: *now*).
    """

    admin_id: str
    emri: str
    mbiemri: str
    email: str
    fjalekalimi: str
    nivel_privilegjeve: int
    gjendja_llogarise: str
    date_regjistrimi: datetime = field(default_factory=datetime.now)

    # ── Actor abstract property ───────────────────────────────────────────────
    @property
    def roli(self) -> str:
        return "Admin"

    # ── Auth behaviours ───────────────────────────────────────────────────────
    def login(self, email: str, password: str) -> bool:
        return self.email == email and self.fjalekalimi == password

    def logout(self) -> None:
        print(f"Admin {self.emri} doli.")

    def verifiko_2fa(self, kod: str) -> bool:
        return kod is not None and len(kod) == 6

    # ── Domain behaviours ─────────────────────────────────────────────────────
    def blok_ip(self, ip: str) -> None:
        print(f"IP bllokuar: {ip}")

    def riaktivizo_llogari(self, user_id: str) -> None:
        print(f"Llogaria riaktivizuar: {user_id}")

    def konfiguroj_firewall(self) -> None:
        print("Firewall konfiguruar.")

    def gjeneroj_raport(self) -> None:
        print("Raport gjeneruar.")

    def nivel_label(self) -> str:
        """Return a human-readable privilege-level label."""
        return {1: "ReadOnly", 2: "Standard", 3: "Superadmin"}.get(
            self.nivel_privilegjeve, "Unknown"
        )

    # ── Dunder methods ────────────────────────────────────────────────────────
    def __str__(self) -> str:
        return (
            f"[{self.admin_id}] {self.emri} {self.mbiemri}"
            f" — Nivel {self.nivel_privilegjeve}"
        )

    def __repr__(self) -> str:
        return (
            f"AdminSistemi(admin_id={self.admin_id!r}, emri={self.emri!r},"
            f" mbiemri={self.mbiemri!r}, email={self.email!r},"
            f" nivel_privilegjeve={self.nivel_privilegjeve!r},"
            f" gjendja_llogarise={self.gjendja_llogarise!r})"
        )

    def __eq__(self, other: object) -> bool:
        if not isinstance(other, AdminSistemi):
            return NotImplemented
        return self.admin_id == other.admin_id

    def __hash__(self) -> int:
        return hash(self.admin_id)
